2017-11-01  Dinkar Khattar  <dinkar@lnxsrv05.seas.ucla.edu>

  * lib/timespec.h (timespec_cmp):
  We notice that this function uses integer subtraction, which is prone to integer overflow.
  In some cases, the difference between a and b is so high that we get inaccurate values.
  This patch uses 3 defualt values of diff based on the comparision between a and b.

--- coreutils-with-bug/lib/timespec.h	2017-11-01 10:07:38.206468000 -0700
+++ timespec.h	2017-11-01 10:13:22.545488000 -0700
@@ -45,8 +45,10 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int diff = 0;
+  if (a.tv_sec > b.tv_sec) {diff = 1;}
+  else if (a.tv_sec < b.tv_sec) {diff = -1;}
+  return diff;
 }

 # if ! HAVE_DECL_NANOSLEEP
